### PR TITLE
Added logic to sync cluster network details

### PR DESCRIPTION
**Pull request by shtripat**
tendrl-bug-id: Tendrl/ceph-integration#310
Signed-off-by: Shubhendu <shtripat@redhat.com>

### CODE REVIEW

**Comment by shtripat** [comment]
Not yet verified

File: tendrl/ceph_integration/sds_sync/__init__.py [hunk 1]
```diff
@@ -179,6 +182,25 @@ def on_heartbeat(self, cluster_data):
...
+        try:
+            NS.tendrl.objects.ClusterNetworkDetails(
+                integration_id=NS.tendrl_context.integration_id
+            ).load()
+        except etcd.EtcdKeyNotFound:
+            try:
+                cluster_config = NS.ceph.objects.SyncObject(
+                    sync_type='config'
+                ).load().data
+                NS.tendrl.objects.ClusterNetworkDetails(
```
Comment (r0h4n) [inline tendrl/ceph_integration/sds_sync/__init__.py]: ClusterNetworkDetails object should be `Cluster` 

check my comments here https://github.com/Tendrl/commons/pull/631

**Review by r0h4n — CHANGES_REQUESTED** [review]

File: tendrl/ceph_integration/sds_sync/__init__.py (hunk 1 above)
Comment (shtripat) [reply tendrl/ceph_integration/sds_sync/__init__.py]: Done

File: tendrl/ceph_integration/sds_sync/__init__.py [hunk 2]
```diff
@@ -179,6 +182,26 @@ def on_heartbeat(self, cluster_data):
         # Sync the OSD utilization details
         self._sync_osd_utilization()
 
+    def _sync_cluster_network_details(self):
+        try:
+            NS.tendrl.objects.ClusterNetworkDetails(
```
Comment (r0h4n) [inline tendrl/ceph_integration/sds_sync/__init__.py]: use NS.tendrl.objects.Cluster

File: tendrl/ceph_integration/sds_sync/__init__.py [hunk 3]
```diff
@@ -179,6 +182,26 @@ def on_heartbeat(self, cluster_data):
...
+                integration_id=NS.tendrl_context.integration_id
+            ).load()
```
Comment (r0h4n) [inline tendrl/ceph_integration/sds_sync/__init__.py]: use exists instead of load, then you dont need the try/except


update : IGNORE this comment

**Review by r0h4n — CHANGES_REQUESTED** [review]

File: tendrl/ceph_integration/sds_sync/__init__.py [hunk 4]
```diff
@@ -179,6 +182,26 @@ def on_heartbeat(self, cluster_data):
...
+                ).load().data
+                cluster = NS.tendrl.objects.Cluster(
+                    integration_id=NS.tendrl_context.integration_id
+                ).load()
+                cluster.public_network = cluster_config['public_network']
+                cluster.cluster_network = cluster_config['cluster_network']
+                cluster.save()
+            except etcd.EtcdKeyNotFound:
+                # Pass as next time it should try to sync the network details
+                pass
```
Comment (r0h4n) [inline tendrl/ceph_integration/sds_sync/__init__.py]: log appropriate msg and raise this exception instead of pass

**Review by r0h4n — CHANGES_REQUESTED** [review]

File: tendrl/ceph_integration/sds_sync/__init__.py (hunk 2 above)
Comment (shtripat) [reply tendrl/ceph_integration/sds_sync/__init__.py]: Ah my bad. It should be like done in gluster-integration. will change it.

File: tendrl/ceph_integration/sds_sync/__init__.py (hunk 4 above)
Comment (shtripat) [reply tendrl/ceph_integration/sds_sync/__init__.py]: Done

**Review by r0h4n — APPROVED** [review]

**Pull request merged**